cmake/sysbuild/image_signing_extra.cmake

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+if(NOT SB_CONFIG_BOOTLOADER_MCUBOOT)
+  return()
+endif()
+
+if(NOT SB_CONFIG_MCUBOOT_EXTRA_IMAGES OR SB_CONFIG_MCUBOOT_EXTRA_IMAGES EQUAL 0)
+  return()
+endif()
+
+get_property(extra_paths GLOBAL PROPERTY DFU_EXTRA_PATHS)
+if(NOT extra_paths)
+  return()
+endif()
+
+set(keyfile "${SB_CONFIG_BOOT_SIGNATURE_KEY_FILE}")
+set(keyfile_enc "${SB_CONFIG_BOOT_ENCRYPTION_KEY_FILE}")
+string(CONFIGURE "${keyfile}" keyfile)
+string(CONFIGURE "${keyfile_enc}" keyfile_enc)
+
+if(NOT "${SB_CONFIG_BOOT_SIGNATURE_TYPE_NONE}")
+  if("${keyfile}" STREQUAL "")
+    message(WARNING "Neither SB_CONFIG_BOOT_SIGNATURE_TYPE_NONE or "
+                    "SB_CONFIG_BOOT_SIGNATURE_KEY_FILE are set, the generated build will not be "
+                    "bootable by MCUboot unless it is signed manually/externally.")
+    return()
+  endif()
+endif()
+
+if(NOT DEFINED IMGTOOL)
+  message(FATAL_ERROR "Can't sign images for MCUboot: can't find imgtool. To fix, install imgtool with pip3, or add the mcuboot repository to the west manifest and ensure it has a scripts/imgtool.py file.")
+  return()
+endif()
+
+get_property(extra_names GLOBAL PROPERTY DFU_EXTRA_NAMES)
+get_property(extra_versions GLOBAL PROPERTY DFU_EXTRA_VERSIONS)
+get_property(extra_targets GLOBAL PROPERTY DFU_EXTRA_TARGETS)
+
+# Iterate through all extra images and sign them
+list(LENGTH extra_paths num_binaries)
+if(num_binaries GREATER 0)
+  math(EXPR last_index "${num_binaries} - 1")
+
+  set(signed_paths "")
+  set(signed_names "")
+
+  foreach(index RANGE ${last_index})
+    math(EXPR slot_number "${index} + 1")
+
+    if(NOT DEFINED NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER OR NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER EQUAL -1)
+      message(FATAL_ERROR "Extra image slot ${slot_number} not configured. Set SB_CONFIG_MCUBOOT_EXTRA_IMAGES to at least ${num_binaries} in sysbuild.")
+    endif()
+
+    set(image_id ${NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER})
+
+    list(GET extra_paths ${index} input_bin)
+    list(GET extra_names ${index} image_name)
+    list(GET extra_versions ${index} version)
+
+    if(NOT image_name OR image_name STREQUAL "")
+      get_filename_component(image_name "${input_bin}" NAME_WE)
+    endif()
+
+    set(output_bin "${CMAKE_BINARY_DIR}/${image_name}.signed.bin")
+    set(output_hex "${CMAKE_BINARY_DIR}/${image_name}.signed.hex")
+
+    set(imgtool_sign 
+      ${PYTHON_EXECUTABLE} ${IMGTOOL} sign
+      --version ${version}
+      --align 16
```

Comment:
align 16 would only be for nrf54l15, see https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/mcuboot.cmake#L132 for how to get the align size